### DISCUSSION TITLE
"TypeError: Failed to construct 'HTMLElement': Illegal constructor" while trying to bootstrap a project using EdgelessEditor in Preset with Vue

### DISCUSSION THREAD

**Discussion by khangho-vgames** [toeverything/blocksuite#8927] [Q&A] [answered]
Hi,

I have an `editor.ts` file like this:

```ts
import { createEmptyDoc, EdgelessEditor } from "@blocksuite/presets";

export function initEditor() {
  const doc = createEmptyDoc().init();
  const editor = new EdgelessEditor();
  editor.doc = doc;
  return { editor };
}
```

and then I will try to call it and init it in the Container like so:

```vue
<template>
  <div class="editor-container" ref="editorContainerRef"></div>
</template>

<script setup lang="ts">
import { inject, onMounted, ref } from "vue";
import { initEditor } from "../editor.ts";

const { editor } = initEditor();
const editorContainerRef = ref<HTMLDivElement>();

onMounted(() => {
  editorContainerRef.value && editorContainerRef.value.appendChild(editor);
});
</script>
```

It will throw the above error like this:

![{D1757FDE-664F-490E-A2F2-6D944C57A9F0}](https://github.com/user-attachments/assets/7b81bb38-799c-41da-8109-398bd1e71194)

I'm using the latest package from @blocksuite:
```json
{
 "dependencies": {
    "@blocksuite/affine": "^0.18.5",
    "@blocksuite/blocks": "^0.18.5",
    "@blocksuite/presets": "^0.18.5",
    "vue": "3.4.27"
  },
}
```

I was testing with the 0.15.0-canary version before, it works fine. But I can't bootstrap with 0.18.x

Can someone help?
The document is outdated and don't have any helpful guide on this.

**Comment by OpenBitSack** [comment, marked as answer]
import {effects as blocksEffects } from '@blocksuite/blocks/effects'
import {effects as presetsEffects} from '@blocksuite/presets/effects'

blocksEffects()
presetsEffects()

You should add this code before initializing editor

**Reply by khangho-vgames** [reply]
Thanks, it works.

But in a bit further code digging. I found that the `effects()` function imported from `@blocksuite/affine/effects` call both of the block and presets effects:

```ts
// @blocksuite\affine\src\effects.ts

import { effects as blocksEffects } from '@blocksuite/blocks/effects';
import { effects as presetsEffects } from '@blocksuite/presets/effects';

export function effects() {
  blocksEffects();
  presetsEffects();
}
```

So if anyone encounter the same problem, can call this one only.

**Reply by khangho-vgames** [reply]
![{68FEB63E-780E-45D9-B5C0-FFF39B046579}](https://github.com/user-attachments/assets/859dfc41-f0ed-4207-82cc-1da69597b50c)

but one more problem is even the editor is working now. The styling is lacking? I guess that is another initiator function need to be called?

**Reply by khangho-vgames** [reply]
Found the fix for the missing style:

- Need to install the `@toeverything/themes` package.
- `import "@toeverything/theme/style.css";` in your project

It feels weird that we need to install another package for a "preset" editor but I think they're still in rapid development so these QoL features are missing is understandable.

**Reply by mingfang** [reply]
Thanks for sharing the workaround.  
I searched for hours before finding this.
IMO this is a bug and should be fixed in the framework; we should not have to apply this workaround.
